src/utils/hwWallet.js

```diff
@@ -10,37 +10,57 @@ import { getTransactionBytes, calculateTxId, getBufferToHex } from './rawTransac
 import { PLATFORM_TYPES, getPlatformType } from './platform';
 import store from '../store';
 import loginTypes from '../constants/loginTypes';
+import { accountLoggedOut } from '../actions/account';
+import { dialogDisplayed, dialogHidden } from '../actions/dialog';
+import Alert from '../components/dialog/alert';
 
 const util = require('util');
 
 const { ipc } = window;
 
+const handleConnect = () => {
+  store.dispatch({
```

Comment:
I know it's not new in this PR, but IMO we shouldn't use `import store from '../store';`, because it causes lot of problems with unit tests when it gets imported in completely unrelated unit tests.

So far, the redux architecture we used was that if something needs to `dispatch` an action, it should be a `middleware`, and utils should be redux-independent.

Reply:
ok Vit right now I just put the login in the login middleware file so right now it is better